bus-mapping/src/circuit_input_builder.rs

```diff
@@ -286,6 +286,47 @@ impl<'a, C: CircuitsParams> CircuitInputBuilder<C> {
         }
     }
 
+    fn check_and_chunk(
+        &mut self,
+        geth_trace: &GethExecTrace,
+        tx: &'a mut Transaction,
+        tx_ctx: &'a mut TransactionContext,
```

Comment:
i use ref cuz in this function the ownership is being moved multiple times, so i have to clone from outside ... but anyways i took ur suggestion :) 
https://github.com/CeciliaZ030/zkevm-circuits/blob/11fa0629497d96da9b85dbed32a11a6381846684/bus-mapping/src/circuit_input_builder.rs#L377

Reply:
nah maybe i'll keep the old way... because of `let mut cib_ref = cib.state_ref(&mut tx, &mut tx_ctx_);` we have error[E0502]: cannot borrow `tx_ctx` as immutable because it is also borrowed as mutable, so i have to call clone again anyways, may as well just use mutable ref so that the caller function has fewer clones 😅

Reply:
My old suggestion way should work, kindly check code diff below :)
https://github.com/CeciliaZ030/zkevm-circuits/compare/bus-mapping-chunk-...hero78119:sm/example1?expand=1